jwt/jwt.c

```diff
@@ -83,21 +85,34 @@ char* jwt_create(char * project_id, long long int time, NN_DIGIT * priv_key, int
 
 int jwt_verify(char * jwt,  point_t * pub_key)
 {
-    ecc_init(); // TODO: Has to be initialized only once -> move to broker init phase
-
-    char * chunk_head = jwt;
-    char * dot_pos = strchr(chunk_head, '.');
-    simple_array_t jwt_header = base64_decode(chunk_head, dot_pos - chunk_head + 1);
-    chunk_head = dot_pos + 1;
-    dot_pos = strchr(chunk_head, '.');
-    simple_array_t jwt_body = base64_decode(chunk_head, dot_pos - chunk_head + 1);
-    chunk_head = dot_pos + 1;
-    simple_array_t jwt_signature = base64_decode(chunk_head, strlen(chunk_head));
+    //ecc_init(); // TODO: Has to be initialized only once -> move to broker init phase
+
+    // First check if jwt is in correct format then decode
+    char * dot_pos_1 = strchr(jwt, '.');
+    if (!dot_pos_1) {
+        mosquitto_log_printf(MOSQ_LOG_ERR, "Invalid JWT: Missing dot!");
+        return 0;
+    }  // dot_pos_1 = header_end
+    
+    char * dot_pos_2 = strchr(dot_pos_1 + 1, '.');
+    if (!dot_pos_2) {
+        mosquitto_log_printf(MOSQ_LOG_ERR, "Invalid JWT: Missing dot!");
+        return 0;
+    }  // dot_pos_2 = body_end
+
+    if (strlen(dot_pos_2 + 1) < (2 * NUMBYTES)) {
```

Comment:
Why do you compare with  2 * NUMBYTES?

Reply:
Fist I was comparing with 64 but then I thought I should compare it by 2 * NUMBYTES as signature consists of signature_r and signature_s. Both worked.

Reply:
Well okay, it's good enough.  base64 encoding increases the size by around 35%, so it definitely can not be less than the raw data.